packages/nimble-components/src/table/specs/table-column-specs/table-column-menu-button.md

```diff
@@ -0,0 +1,205 @@
+    </nimble-table-column-menu-button>
+
+    <nimble-menu slot="software-version-menu">
+        <nimble-menu-item>1.0.0</nimble-menu-item>
+        <nimble-menu-item>1.5.0</nimble-menu-item>
+        <nimble-menu-item>2.0.0</nimble-menu-item>
+    </nimble-menu>
+</nimble-table>
+```
+
+### API
+
+#### Column Component
+
+_Component Name_
+
+-   `nimble-table-column-menu-button`
+
+_Props/Attrs_
+
+-   `field-name`: string - The name of the field in each record that contains the string that will be displayed in the menu button
+-   `menu-slot`: string - The name of the slot within the `nimble-table` instance where the menu associated with the column's menu button will be provided
+-   `fractional-width`: number (defaults to 1)
+-   `min-pixel-width`: number (defaults to minimum supported by table)
+
+_Content_
+
+-   column title (icon and/or text)
+
+_Events_
+
+-   `menu-button-beforetoggle`: Fired when a menu button in a cell fires the `beforetoggle` event
+-   `menu-button-toggle`: Fired when a menu button in a cell fires the `toggle` event
+
+The detail for each of these events will be as follows:
+
+```ts
+interface MenuButtonColumnToggleEventDetail {
+    recordId: string;
+    newState: boolean;
+    oldState: boolean;
+    originalEvent: CustomEvent<MenuButtonToggleEventDetail>;
+}
+```
+
+#### Cell View Component
+
+_Component Name_
+
+-   `nimble-table-column-menu-button-cell-view`
+
+_Rendering_
+
+If the record value associated with the specified `field-name` is a non-empty string, the cell will render a menu button containing the record value text. If the record value associated with the specified `field-name` is not a non-empty string, the cell will render nothing.
+
+The menu button will have the following behaviors/styles:
+
+-   Use `ghost` appearance mode
+-   Have `nimble-icon-arrow-expander-down` slotted into the `end` slot
+-   Grow to fill the width of the cell
+-   Left-align the text (with padding) in the button to make the text within each button is aligned for a given column
+    -   By default, button text is centered, so custom styling will be written to handle this case.
+-   Ellipsize text in the button if it doesn't fit in the button
+-   Provide the full text within the button's `title` if the text is ellipsized
+    -   It is unlikely this can leverage the existing `overflow` directive as-is because the overflow will occur on the text within a `span` within the button, but the `title` should be added when the button is hovered. The best solution to this problem will be determined during implementation.
+
+#### Group Header View Component
+
+A new component will not be created for the group header view of the menu button column. The column will specify the existing `tableColumnTextGroupHeaderViewTag` as the `groupHeaderViewTag` because the header will contain only text.
+
+### Sorting & Grouping
+
+The column will be sorted and grouped based on the text that is provided for display in the button.
```

Comment:
Yes, I would expect that changing the value through the menu would change its positioning in the grid (assuming it didn't fall in the same place alphabetically anymore).

However, all nimble-table columns support disabling sorting via the `sorting-disabled` attribute. I don't think we've exposed that configuration on our existing sl-table columns because we haven't had a use case for that capability on any of the columns yet. But, I don't foresee an issue with making that a configuration on the sl-table column wrapping this nimble column. If we felt strongly that this was going to be the most common use-case in SLE, we could even default the column to not allow sorting and make an app opt-in to sorting.